proptest/src/num/float_samplers.rs

```diff
@@ -0,0 +1,338 @@
+//-
+// Copyright 2022 Michael Wright
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Alternative uniform float samplers because the ones provided by the rand crate are prone
+//! to overflow. The samplers work by uniformly selecting from a set of equally spaced values in
+//! the interval and the included bounds. Selection is slighly biased towards the bounds.
+
+pub(crate) use self::f32::F32U;
+pub(crate) use self::f64::F64U;
+
+macro_rules! float_sampler {
```

Comment:
Because of the subtle nuances in this code, i highly suggest a comment about the overall strategy with an example of what's expected in terms of sampling given some range.